test/extensions/filters/http/cache/http_cache_utils_test.cc

```diff
@@ -0,0 +1,83 @@
+#include <vector>
+
+#include "common/http/header_map_impl.h"
+
+#include "extensions/filters/http/cache/http_cache_utils.h"
+
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+namespace {
+
+// TODO(toddmgreer) Add tests for eat* functions
+// TODO(toddmgreer) More tests for httpTime, effectiveMaxAge
+
+class HttpTimeTest : public testing::TestWithParam<const char*> {
+protected:
+  Http::TestHeaderMapImpl response_headers_{{"date", GetParam()}};
+};
+
+const char* const ok_times[] = {
+    "Sun, 06 Nov 1994 08:49:37 GMT",  // IMF-fixdate
+    "Sunday, 06-Nov-94 08:49:37 GMT", // obsolete RFC 850 format
+    "Sun Nov  6 08:49:37 1994"        // ANSI C's asctime() format
+};
+
+INSTANTIATE_TEST_SUITE_P(Ok, HttpTimeTest, testing::ValuesIn(ok_times));
+
+TEST_P(HttpTimeTest, Ok) {
+  const std::time_t time = SystemTime::clock::to_time_t(Utils::httpTime(response_headers_.Date()));
+  EXPECT_STREQ(ctime(&time), "Sun Nov  6 08:49:37 1994\n");
```

Comment:
can you add tests for the other accepted date formats?

Reply:
They were being tested, but it was hidden in the test fixture. I think it's more clear now.